yarn-project/pxe/src/database/kv_pxe_database.ts

```diff
@@ -282,10 +308,62 @@ export class KVPxeDatabase implements PxeDatabase {
     return result;
   }
 
-  getNotes(filter: NoteFilter): Promise<IncomingNoteDao[]> {
+  getIncomingNotes(filter: IncomingNotesFilter): Promise<IncomingNoteDao[]> {
```

Comment:
Probably because it's also used in estimateSize function and there we don't want to return a promise (also was confused when I first saw it). This is a consequence of us strictly adhering to predefined interface which returns Promise here.

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/7022/commits/fffc176837757d27191a9cbc35b44ca11cb9b27b